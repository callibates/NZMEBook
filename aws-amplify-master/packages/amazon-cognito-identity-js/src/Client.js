import UserAgent from './UserAgent';
/** @class */
export default class Client {
  /**
   * Constructs a new AWS Cognito Identity Provider client object
   * @param {string} region AWS region
   * @param {string} endpoint endpoint
   */
  constructor(region, endpoint) {
    this.endpoint = endpoint || `https://cognito-idp.${region}.amazonaws.com/`;
    this.userAgent = UserAgent.prototype.userAgent || 'aws-amplify/0.1.x js';
  }

  /**
   * Makes an unauthenticated request on AWS Cognito Identity Provider API
   * using fetch
   * @param {string} operation API operation
   * @param {object} params Input parameters
   * @param {function} callback Callback called when a response is returned
   * @returns {void}
  */
  request(operation, params, callback) {
    const headers = {
      'Content-Type': 'application/x-amz-json-1.1',
      'X-Amz-Target': `AWSCognitoIdentityProviderService.${operation}`,
      'X-Amz-User-Agent': this.userAgent,
    };

    const options = {
      headers,
      method: 'POST',
      mode: 'cors',
      cache: 'no-cache',
      body: JSON.stringify(params),
    };

    let response;
    let responseJsonData;

    fetch(this.endpoint, options)
      .then(resp => {
        response = resp;
        return resp;
      })
      .catch(err => {
        // If error happens here, the request failed
        // if it is TypeError throw network error
        if (err instanceof TypeError) {
          throw new Error('Network error');
        }
        throw err;
      })
      .then(resp => resp.json())
      .catch(err => {
        // If error happens here
        // cannot parse the body stream, return undefined
        if (response.ok) return callback(null, undefined);
        else {
          const error = {
            code: response.status, 
            statusCode: response.status,
            message: response.statusText
          }
          callback(error);
        }
      })
      .then(data => {
        // return parsed body stream
        if (response.ok) return callback(null, data);
        responseJsonData = data;

        // Taken from aws-sdk-js/lib/protocol/json.js
        // eslint-disable-next-line no-underscore-dangle
        const code = (data.__type || data.code).split('#').pop();
        const error = {
          code,
          name: code,
          message: (data.message || data.Message || null),
        };
        return callback(error);
      })
      .catch(err => {
        // if cannot split the data
        // default to return 'UnknownError' with the json data from response
        let error = { code: 'UnknownError', message: 'Unknown error, the response body from fetch is: ' + responseJsonData};

        // first check if we have a service error
        if (response && response.headers && response.headers.get('x-amzn-errortype')) {
          try {
            const code = (response.headers.get('x-amzn-errortype')).split(':')[0];
            error = {
              code,
              name: code,
              statusCode: response.status,
              message: (response.status) ? response.status.toString() : null,
            };
          } catch (ex) {
              // pass through so it doesn't get swallowed if we can't parse it
              error = {
                code: 'UnknownError',
                message: response.headers.get('x-amzn-errortype'),
              }
              return callback(error);
          }
        // otherwise check if error is Network error
        } else if (err instanceof Error && err.message === 'Network error') {
          error = {
            code: 'NetworkError',
            name: err.name,
            message: err.message,
          };
        }
        return callback(error);
      });
  }
}
