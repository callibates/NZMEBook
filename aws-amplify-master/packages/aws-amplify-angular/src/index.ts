export { AmplifyAngularModule } from './aws-amplify-angular.module';

export * from './components';
export * from './providers';
