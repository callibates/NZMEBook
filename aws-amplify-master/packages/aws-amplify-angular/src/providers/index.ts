export { AmplifyService } from './amplify.service';
export { AuthState } from './auth.state';
