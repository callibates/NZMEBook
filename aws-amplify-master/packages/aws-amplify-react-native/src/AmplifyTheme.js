/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { StyleSheet } from 'react-native';
export default AmplifyTheme = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'space-around',
        paddingTop: 20,
        width: '100%'
    },
    section: {
        flex: 1,
        width: '100%'
    },
    sectionHeader: {
    },
    sectionHeaderText: {
        width: '100%',
        padding: 10,
        textAlign: 'center',
        backgroundColor: '#007bff',
        color: '#ffffff',
        fontSize: 20,
        fontWeight: '500'
    },
    sectionFooter: {
        width: '100%',
        marginTop: 15,
        padding: 10,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    sectionFooterLink: {
        fontSize: 14,
        color: '#007bff',
        alignItems: 'baseline',
        textAlign: 'center'
    },
    sectionBody: {
    },
    cell: {
        flex: 1,
        width: '50%'
    },
    errorRow: {
        flex: 1
    },

    photo: {
        width: '100%'
    },
    album: {
        width: '100%'
    },

    a: {
    },
    button: {
        backgroundColor: '#007bff'
    },

    input: {
        margin: 6
    }
});

