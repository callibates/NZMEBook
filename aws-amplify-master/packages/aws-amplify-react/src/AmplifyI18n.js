/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const dict = {
    'fr': {
        'Sign In': "Se connecter",
        'Sign Up': "S'inscrire",
        'Sign Out': "Déconnexion",
        'Forgot Password': "Mot de passe oublié",
        'Username': "Nom d'utilisateur",
        'Password': "Mot de passe",
        'Change Password': "Changer le mot de passe",
        'New Password': "nouveau mot de passe",
        'Email': "Email",
        'Phone Number': "Numéro de téléphone",
        'Confirm a Code': "Confirmer un code",
        'Confirm Sign In': "Confirmer la connexion",
        'Confirm Sign Up': "Confirmer l'inscription",
        'Back to Sign In': "Retour à la connexion",
        'Send Code': "Envoyer le code",
        'Confirm': "Confirmer",
        'Resend a Code': "Renvoyer un code",
        'Submit': "Soumettre",
        'Skip': "Sauter",
        'Verify': "Vérifier",
        'Verify Contact': "Vérifier le contact",
        'Code': "Code",
        'Account recovery requires verified contact information':
"La récupération du compte nécessite des informations de contact vérifiées",

        'User does not exist': "L'utilisateur n'existe pas",
        'User already exists': "L'utilisateur existe déjà",
        'Incorrect username or password': "identifiant ou mot de passe incorrect",
        'Invalid password format': "format de mot de passe invalide",
        'Invalid phone number format':
`Format de numéro de téléphone invalide.
Veuillez utiliser un format de numéro de téléphone du +12345678900`
    },

    'es': {
        'Sign In': "Registrarse",
        'Sign Up': "Regístrate",
        'Sign Out': "Desconectar",
        'Forgot Password': "Se te olvidó tu contraseña",
        'Username': "Nombre de usuario",
        'Password': "Contraseña",
        'Change Password': "Cambia la contraseña",
        'New Password': "Nueva contraseña",
        'Email': "Email",
        'Phone Number': "Número de teléfono",
        'Confirm a Code': "Confirmar un código",
        'Confirm Sign In': "Confirmar inicio de sesión",
        'Confirm Sign Up': "Confirmar Registrarse",
        'Back to Sign In': "Volver a Iniciar sesión",
        'Send Code': "Enviar código",
        'Confirm': "Confirmar",
        'Resend a Code': "Reenviar un código",
        'Submit': "Enviar",
        'Skip': "Omitir",
        'Verify': "Verificar",
        'Verify Contact': "Verificar contacto",
        'Code': "Código",
        'Account recovery requires verified contact information':
"La recuperación de la cuenta requiere información de contacto verificada",

        'User does not exist': "el usuario no existe",
        'User already exists': "El usuario ya existe",
        'Incorrect username or password': "Nombre de usuario o contraseña incorrecta",
        'Invalid password format': "Formato de contraseña inválido",
        'Invalid phone number format':
`Formato de número de teléfono inválido.
Utilice el formato de número de teléfono +12345678900`
    },

    'zh': {
        'Sign In': "登录",
        'Sign Up': "注册",
        'Sign Out': "退出",
        'Forgot Password': "忘记密码",
        'Username': "用户名",
        'Password': "密码",
        'Change Password': "改变密码",
        'New Password': "新密码",
        'Email': "邮箱",
        'Phone Number': "电话",
        'Confirm a Code': "确认码",
        'Confirm Sign In': "确认登录",
        'Confirm Sign Up': "确认注册",
        'Back to Sign In': "回到登录",
        'Send Code': "发送确认码",
        'Confirm': "确认",
        'Resend a Code': "重发确认码",
        'Submit': "提交",
        'Skip': "跳过",
        'Verify': "验证",
        'Verify Contact': "验证联系方式",
        'Code': "确认码",
        'Account recovery requires verified contact information':
"账户恢复需要验证过的联系方式",

        'User does not exist': "用户不存在",
        'User already exists': "用户已经存在",
        'Incorrect username or password': "用户名或密码错误",
        'Invalid password format': "密码格式错误",
        'Invalid phone number format': "电话格式错误，请使用格式 +12345678900"
    }
}

export default dict;
