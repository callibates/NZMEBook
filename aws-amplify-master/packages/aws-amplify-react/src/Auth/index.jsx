/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Authenticator from './Authenticator';

export { default as Authenticator } from './Authenticator';
export { default as AuthPiece } from './AuthPiece';
export { default as SignIn } from './SignIn';
export { default as ConfirmSignIn } from './ConfirmSignIn';
export { default as SignOut } from './SignOut';
export { default as RequireNewPassword } from './RequireNewPassword';
export { default as SignUp } from './SignUp';
export { default as ConfirmSignUp } from './ConfirmSignUp';
export { default as VerifyContact } from './VerifyContact';
export { default as ForgotPassword } from './ForgotPassword';
export { default as Greetings } from './Greetings';
export { default as FederatedSignIn, FederatedButtons } from './FederatedSignIn';
export { default as TOTPSetup } from './TOTPSetup';

export * from './Provider';

import Greetings from './Greetings';

export function withAuthenticator(Comp, includeGreetings=false, authenticatorComponents = []) {
    return class extends Component {
        constructor(props) {
            super(props);

            this.handleAuthStateChange = this.handleAuthStateChange.bind(this);

            this.state = {
                authState: props.authState || null,
                authData: props.authData || null
            };
        }

        handleAuthStateChange(state, data) {
            this.setState({ authState: state, authData: data });
        }

        render() {
            const { authState, authData } = this.state;
            const signedIn = (authState === 'signedIn');
            if (signedIn) {
                return (
                    <div>
                        {
                            includeGreetings?
                            <Greetings
                                authState={authState}
                                authData={authData}
                                onStateChange={this.handleAuthStateChange}
                            />
                            : null
                        }
                        <Comp
                            {...this.props}
                            authState={authState}
                            authData={authData}
                            onStateChange={this.handleAuthStateChange}
                        />
                    </div>
                )
            }

            return <Authenticator
                {...this.props}
                hideDefault={authenticatorComponents.length > 0}
                onStateChange={this.handleAuthStateChange}
                children={authenticatorComponents}
            />
        }
    }
}

export class AuthenticatorWrapper extends Component {
    constructor(props) {
        super(props);

        this.state = { auth: 'init' };

        this.handleAuthState = this.handleAuthState.bind(this);
    }

    handleAuthState(state, data) {
        this.setState({ auth: state, authData: data });
    }

    render() {
        return (
            <div>
                <Authenticator {...this.props} onStateChange={this.handleAuthState} />
                {this.props.children(this.state.auth)}
            </div>
        )
    }
}

AuthenticatorWrapper.propTypes = {
    children: PropTypes.func.isRequired
}
