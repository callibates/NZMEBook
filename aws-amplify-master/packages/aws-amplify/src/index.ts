/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import Analytics, { AnalyticsClass, AnalyticsProvider, AWSPinpointProvider, AWSKinesisProvider } from './Analytics';
import Auth, { AuthClass } from './Auth';
import Storage, { StorageClass } from './Storage';
import API, { APIClass, graphqlOperation } from './API';
import PubSub, { PubSubClass } from './PubSub';
import Cache from './Cache';
import Interactions, { InteractionsClass } from './Interactions';
import {
    ConsoleLogger as Logger,
    Hub,
    JS,
    ClientDevice,
    Signer,
    I18n,
    Amplify,
    ServiceWorker
} from './Common';

export default Amplify;

Amplify.Auth = Auth;
Amplify.Analytics = Analytics;
Amplify.API = API;
Amplify.Storage = Storage;
Amplify.I18n = I18n;
Amplify.Cache = Cache;
Amplify.PubSub = PubSub;
Amplify.Logger = Logger;
Amplify.ServiceWorker = ServiceWorker;
Amplify.Interactions = Interactions;

export { 
    Auth, 
    Analytics, 
    Storage,
    API, 
    PubSub, 
    I18n, 
    Logger, 
    Hub, 
    Cache, 
    JS, 
    ClientDevice, 
    Signer, 
    ServiceWorker, 
    Interactions 
};

export { 
    AuthClass, 
    AnalyticsClass, 
    APIClass, 
    StorageClass,
    PubSubClass,
    InteractionsClass,
    AnalyticsProvider, 
    AWSPinpointProvider, 
    AWSKinesisProvider 
};
export { graphqlOperation };
