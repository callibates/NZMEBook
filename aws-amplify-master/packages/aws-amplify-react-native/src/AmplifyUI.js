/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import { View, Text, TextInput, TouchableHighlight } from 'react-native';

import { I18n } from 'aws-amplify';

export const Username = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Username')}
            autoFocus={true}
            autoCapitalize="none"
            {...props}
        />
    )
}

export const Password = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Password')}
            secureTextEntry={true}
            {...props}
        />
    )
}

export const Email = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Email')}
            keyboardType="email-address"
            autoCapitalize="none"
            {...props}
        />
    )
}

export const PhoneNumber = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Phone Number')}
            keyboardType="phone-pad"
            {...props}
        />
    )
}

export const ConfirmationCode = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Code')}
            autoFocus={true}
            {...props}
        />
    )
}

export const LinkCell = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.cell}>
            <TouchableHighlight
                onPress={props.onPress}
            >
                <Text style={theme.sectionFooterLink}>{props.children}</Text>
            </TouchableHighlight>
        </View>
    )
}

export const Header = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.sectionHeader}>
            <Text style={theme.sectionHeaderText}>{props.children}</Text>
        </View>
    )
}

export const ErrorRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.errorRow}>
            <Text style={theme.erroRowText}>{props.children}</Text>
        </View>
    )
}
