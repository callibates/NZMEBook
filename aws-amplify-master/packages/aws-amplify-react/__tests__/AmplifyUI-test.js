import React from 'react';
import * as UI from '../src/AmplifyUI';

describe('AmplifyUi test', () => {
    test('render InputRow correctly', () => {
        const wrapper = shallow(<UI.InputRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render RadioRow correctly', () => {
        const wrapper = shallow(<UI.RadioRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render CheckboxRow correctly', () => {
        const wrapper = shallow(<UI.CheckboxRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render MessageRow correctly', () => {
        const wrapper = shallow(<UI.MessageRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render ButtonRow correctly', () => {
        const wrapper = shallow(<UI.ButtonRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Link correctly', () => {
        const wrapper = shallow(<UI.Link/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Container correctly', () => {
        const wrapper = shallow(<UI.Container/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render FormContainer correctly', () => {
        const wrapper = shallow(<UI.FormContainer/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render FormSection correctly', () => {
        const wrapper = shallow(<UI.FormSection/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render ErrorSection correctly', () => {
        const wrapper = shallow(<UI.ErrorSection/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SectionHeader correctly', () => {
        const wrapper = shallow(<UI.SectionHeader/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SectionHeaderContent correctly', () => {
        const wrapper = shallow(<UI.SectionHeaderContent/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SectionFooter correctly', () => {
        const wrapper = shallow(<UI.SectionFooter/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SectionFooterContent correctly', () => {
        const wrapper = shallow(<UI.SectionFooterContent/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SectionBody correctly', () => {
        const wrapper = shallow(<UI.SectionBody/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render ActionRow correctly', () => {
        const wrapper = shallow(<UI.ActionRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render FormRow correctly', () => {
        const wrapper = shallow(<UI.FormRow/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Radio correctly', () => {
        const wrapper = shallow(<UI.Radio/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Checkbox correctly', () => {
        const wrapper = shallow(<UI.Checkbox/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Button correctly', () => {
        const wrapper = shallow(<UI.Button/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render ButtonContent correctly', () => {
        const wrapper = shallow(<UI.ButtonContent/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render SignInButton correctly', () => {
        const wrapper = shallow(<UI.SignInButton/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Label correctly', () => {
        const wrapper = shallow(<UI.Label/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Space correctly', () => {
        const wrapper = shallow(<UI.Space/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render NavBar correctly', () => {
        const wrapper = shallow(<UI.NavBar/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render Nav correctly', () => {
        const wrapper = shallow(<UI.Nav/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render NavRight correctly', () => {
        const wrapper = shallow(<UI.NavRight/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render NavItem correctly', () => {
        const wrapper = shallow(<UI.NavItem/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });

    test('render NavButton correctly', () => {
        const wrapper = shallow(<UI.NavButton/>);
        wrapper.setProps({ theme: 'theme' });
        expect(wrapper).toMatchSnapshot();
    });
});