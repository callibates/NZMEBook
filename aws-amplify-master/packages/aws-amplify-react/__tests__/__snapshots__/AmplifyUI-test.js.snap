// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmplifyUi test render ActionRow correctly 1`] = `
<div
  className="amplify-action-row"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render Button correctly 1`] = `
<button
  className="amplify-button"
  style={Object {}}
>
  <ButtonContent
    theme="theme"
  />
</button>
`;

exports[`AmplifyUi test render ButtonContent correctly 1`] = `
<span
  className="amplify-button-content"
/>
`;

exports[`AmplifyUi test render ButtonRow correctly 1`] = `
<div
  className="amplify-action-row"
>
  <Button
    theme="theme"
  />
</div>
`;

exports[`AmplifyUi test render Checkbox correctly 1`] = `
<input
  className="amplify-checkbox"
  style={Object {}}
  type="checkbox"
/>
`;

exports[`AmplifyUi test render CheckboxRow correctly 1`] = `
<FormRow
  theme="theme"
>
  <Checkbox
    id="_undefined"
    theme="theme"
  />
  <Label
    htmlFor="_undefined"
    theme="theme"
  />
</FormRow>
`;

exports[`AmplifyUi test render Container correctly 1`] = `
<div
  className="amplify-container"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render ErrorSection correctly 1`] = `
<div
  className="amplify-error-section"
  style={Object {}}
>
  <ErrorSectionContent />
</div>
`;

exports[`AmplifyUi test render FormContainer correctly 1`] = `
<div
  className="amplify-form-container"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render FormRow correctly 1`] = `
<div
  className="amplify-form-row"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render FormSection correctly 1`] = `
<FormContainer
  theme="theme"
>
  <div
    className="amplify-form-section"
    style={Object {}}
  />
</FormContainer>
`;

exports[`AmplifyUi test render InputRow correctly 1`] = `
<FormRow
  theme="theme"
>
  <input
    className="amplify-input"
    style={Object {}}
  />
</FormRow>
`;

exports[`AmplifyUi test render Label correctly 1`] = `
<label
  className="amplify-label"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render Link correctly 1`] = `
<a
  className="amplify-a"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render MessageRow correctly 1`] = `
<FormRow
  theme="theme"
>
  <MessageContent
    theme="theme"
  />
</FormRow>
`;

exports[`AmplifyUi test render Nav correctly 1`] = `
<div
  className="amplify-nav"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render NavBar correctly 1`] = `
<div
  className="amplify-nav-bar"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render NavButton correctly 1`] = `
<button
  className="amplify-nav-button"
  style={Object {}}
>
  <span />
</button>
`;

exports[`AmplifyUi test render NavItem correctly 1`] = `
<div
  className="amplify-nav-item"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render NavRight correctly 1`] = `
<div
  className="amplify-nav-right"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render Radio correctly 1`] = `
<input
  className="amplify-radio"
  style={Object {}}
  type="radio"
/>
`;

exports[`AmplifyUi test render RadioRow correctly 1`] = `
<FormRow
  theme="theme"
>
  <Radio
    id="_undefined"
    theme="theme"
  />
  <Label
    htmlFor="_undefined"
    theme="theme"
  />
</FormRow>
`;

exports[`AmplifyUi test render SectionBody correctly 1`] = `
<div
  className="amplify-section-body"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render SectionFooter correctly 1`] = `
<div
  className="amplify-section-footer"
  style={Object {}}
>
  <SectionFooterContent />
</div>
`;

exports[`AmplifyUi test render SectionFooterContent correctly 1`] = `
<span
  className="amplify-section-footer-content"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render SectionHeader correctly 1`] = `
<div
  className="amplify-section-header"
  style={Object {}}
>
  <SectionHeaderContent
    theme="theme"
  />
</div>
`;

exports[`AmplifyUi test render SectionHeaderContent correctly 1`] = `
<span
  className="amplify-section-header-content"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render SignInButton correctly 1`] = `
<button
  className="amplify-signin-button"
  style={Object {}}
/>
`;

exports[`AmplifyUi test render Space correctly 1`] = `
<span
  className="amplify-space"
  style={Object {}}
/>
`;
